projects/wonka-ui/src/lib/charts/chart.component.ts

```diff
@@ -45,71 +35,187 @@ export interface ChartItem {
+  scatterChartOptions: ChartOptions;
+  scatterChartColors: { backgroundColor: string[]; }[];
+  scatterChartType: ChartType;
 
   constructor() { }
 
 
   ngOnInit() {
+    if (!this.baseColors) {
+      this.baseColors = [[0, 60, 101], [34, 167, 229], [164, 194, 31], [245, 211, 15], [255, 130, 26], [242, 51, 24]];
+    }
+
     if (!this.class) {
       this.class = '';
     }
-    this.chartType = this.chart.chartType;
-    this.chartOptions.title.text =  this.chart.title;
-    if (this.chart.colors) {
-      this.chartColors = this.chart.colors;
+
+    if (this.pieChartItem) {
+      this.createPieChart();
     }
-    this.datasets = [];
-    this.chartOptions.scales.xAxes.push({
-      scaleLabel: {
-         display: this.chart.xLabel ? true : false,
-         labelString: this.chart.xLabel ? this.chart.xLabel : '',
+    if (this.lineChartItem) {
+      this.createLineChart();
+    }
+    if (this.scatterChartItem) {
+      this.createScatterChart();
+    }
+  }
+
+  private createLineChart() {
+    this.lineChartData = this.lineChartItem.chartDataSets;
+    this.lineChartLabels = this.lineChartItem.chartLabels;
+    this.lineChartOptions = {
+      responsive: true,
+      scales: {
+        // We use this empty structure as a placeholder for dynamic theming.
+        xAxes: [{}],
+        yAxes: [
+          {
+            id: 'y-axis-0',
+            position: 'left',
+          }
+        ]
       },
-      type: this.chart.xAxesType
-    });
-    this.chartOptions.scales.yAxes.push({
-      scaleLabel: {
-         display: this.chart.yLabel ? true : false,
-         labelString: this.chart.yLabel ? this.chart.yLabel : '',
+      annotation: {
+        annotations: [
+          {
+            type: 'line',
+            mode: 'vertical',
+            scaleID: 'x-axis-0',
+            value: 'March',
+            borderColor: 'orange',
+            borderWidth: 2,
+            label: {
+              enabled: true,
+              fontColor: 'orange',
+              content: 'LineAnno'
+            }
+          },
+        ],
       },
-      type: this.chart.yAxesType
-    });
+    };
+    this.lineChartLegend = true;
+    this.lineChartType = 'line';
+    this.lineChartColors = this.createLineChartColor(this.lineChartItem.chartDataSets.length);
+  }
 
-    this.chart.dataset.forEach((data, index) => {
-      this.datasets.push(
+  private createPieChart() {
+    this.pieChartOptions = {
+      responsive: true,
+      legend: {
+        position: 'top',
+      },
+      plugins: {
+        datalabels: {
+          formatter: (value, ctx) => {
+            const label = ctx.chart.data.labels[ctx.dataIndex];
+            return label;
+          },
+        },
+      }
+    };
+    this.pieChartLabels = this.pieChartItem.chartLabels;
+    this.pieChartData = this.pieChartItem.chartData;
+    this.pieChartType = 'pie';
+    this.pieChartLegend = true;
+    if (this.pieChartData.length < (this.baseColors.length * 6) + 1) {
```

Comment:
What if we need more colors than what is specified in this.baseColors?

Reply:
These colors are picked to match each other. If there are more than 6 colors, these 6 base colors will be reused with different transparency, up to 6 transparency levels per color 
(6 x 6 = 36 possible colors/slices). 
If by any chance there is more than 36 slices, the default ng2-charts random colors will take over.